app/src/services/projects.ts

```diff
@@ -236,12 +236,19 @@ const getProjectFilterQuery = (type: ProjectListingTypeFilter): string => {
     }
 }
 
-export const getProjects = async (owner?: string, first = 20, skip = 0, type?: ProjectListingTypeFilter): Promise<ProjectsResult> => {
+export const getProjects = async (
+    owner?: string | null,
+    first = 20,
+    skip = 0,
+    type?: ProjectListingTypeFilter | null,
+    streamId?: string | null, // used to search projects which contain this stream
+): Promise<ProjectsResult> => {
     const theGraphUrl = getGraphUrl()
 
     const ownerFilter = owner != null ? `permissions_: { userAddress: "${owner}", canGrant: true }` : null
     const typeFilter = type != null ? getProjectFilterQuery(type) : null
-    const allFilters = [ownerFilter,typeFilter].filter((filter) => !!filter).join(',')
+    const streamIdFilter = streamId != null ? `streams_contains: ["${streamId}"]` : null
+    const allFilters = [ownerFilter, typeFilter, streamIdFilter].filter((filter) => !!filter).join(',')
```

Comment:
Go for `.filter(Boolean)`.

Reply:
Although I'd much rather collect the filters as I go, meaning

```tsx
const filters = []

if (owner) {
    filters.push(`permissions_: { userAddress: "#{owner}", canGrant: true  }`)
}

// if (somethingelse) { filters.push(…) }
```

You wouldn't have to filter empty ones out.